crates/forge/src/actor.rs

```diff
@@ -151,6 +151,20 @@ impl Message<UpdateContracts> for Worker {
     }
 }
 
+pub struct FetchAbis;
+
+impl Message<FetchAbis> for Worker {
+    type Reply = Vec<ForgeAbi>;
+
+    async fn handle(
+        &mut self,
+        _msg: FetchAbis,
+        _ctx: kameo::message::Context<'_, Self, Self::Reply>,
+    ) -> Self::Reply {
+        self.abis_by_path.clone().into_values().collect()
+    }
+}
```

Comment:
have you tested this with a large number of ABIs?
wondering if this won't be very slow once you build up a large map

Reply:
Tested with around 80 ABIs and fetch time from the UI seemed to be ~100ms

I'm aware `clone` might have performance hits once this gets very large. I was thinking we could move the entire trace parsing logic from the UI to an actor too and that would solve it but I want to have a working prototype first before attempting that refactor

Reply:
sounds like a good idea (both the moving to an actor, and the leaving it for later) :+1: